### PR TITLE
chore: add github codespces plugin to microsite

### PR DESCRIPTION
**Pull request by adityasinghal26**
## Hey, I just made a Pull Request!

Add GitHub Codespaces plugin to Backstage

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: microsite/data/plugins/github-codespaces.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+title: GitHub Codespaces
+author: Aditya Singhal
+authorUrl: https://github.com/adityasinghal26
+category: Development
+description: Integrates GitHub Codespaces for a Backstage component with the Authenticated User.
+documentation: https://github.com/adityasinghal26/backstage-plugins/tree/main/plugins/github-codespaces
+iconUrl: https://github.com/adityasinghal26/backstage-plugins/blob/00c9c00ba9acc3135014d6454ccf04f573195eef/plugins/github-codespaces/images/GitHubLogo.png
+npmPackageName: '@adityasinghal26/plugin-github-codespaces'
```
Comment (awanlin) [inline microsite/data/plugins/github-codespaces.yaml:9]: Just noting that the images don't load when I go here: https://www.npmjs.com/package/@adityasinghal26/plugin-github-codespaces. Also, the content doesn't exactly match your current `README`: https://github.com/adityasinghal26/backstage-plugins/tree/main/plugins/github-codespaces, I only mention this as potential Adopters of your plugin may notice that and feel unsure, package security is very much something people are thinking about and these small differences can bring additional worry.

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for submitting this PR to add your plugin to the directory @adityasinghal26, it's a very helpful one! I've left a single comment. Also, not all the build checks have ran for you so there might be some items from that 👍

File: microsite/data/plugins/github-codespaces.yaml (hunk 1 above)
Comment (adityasinghal26) [reply microsite/data/plugins/github-codespaces.yaml:9]: @awanlin I have updated the latest `README` from GitHub in npmjs package. This also fixes the images loading issue at [https://www.npmjs.com/package/@adityasinghal26/plugin-github-codespaces](https://www.npmjs.com/package/@adityasinghal26/plugin-github-codespaces). Let me know if this resolves the concern.

**Comment by adityasinghal26** [comment]
@awanlin @Rugvip Requesting your feedback on the PR and hoping to get an approval to merge this.

**Comment by awanlin** [comment]
OK, looks like they are running now. Will review again tomorrow 👍

File: microsite/data/plugins/github-codespaces.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+tags:
+  - github
+  - codespaces
+  - development
+  - devcontainers
+addedDate: '2023-12-30'
```
Comment (Rugvip) [inline microsite/data/plugins/github-codespaces.yaml:15]:
```suggestion
addedDate: '2023-12-30'

```

Just a formatting fix

**Review by Rugvip — APPROVED** [review]
Alright, thank you! 👍

**Pull request merged**